webhook_to_fedora_messaging/endpoints/apikey.py

```diff
@@ -0,0 +1,86 @@
+from flask import Blueprint, Flask, request, Response, Request
+from ..database import db
+from ..models.apikey import APIKey
+from ..models.user import User
+from datetime import datetime
+from sqlalchemy_helpers import get_or_create
+from .util import not_found, success, bad_request, created, conflict, validate_request, exclude_from_val
+
+app = Flask(__name__)
+apikey_endpoint = Blueprint("apikey_endpoint", __name__)
+
+@apikey_endpoint.route("/apikey", methods=["POST"])
+def create_apikey():
+    """
+        Used for creating a new service by sending a post request to /service/ path.
+    """
+    
+    session = db.Session()
+    body = request.json
+    
+    
+    user = session.query(User).filter(User.username == body['username']).first()
+    if user is None:
+        return not_found()
+    # Can be different parsing here.
+    valid_till = datetime.strptime(body['valid_till'], "%Y-%m-%d")
+    apikey, is_created = get_or_create(session, APIKey, user_id=user.id, name=body['name'], expiry_date=valid_till)
+    
+    if not is_created:
+        return conflict({'message': 'Key Already Exists'})
+    else:
+        return created({'message': 'Created', 'uuid': apikey.id, 'code': apikey.token})
+        
+    
+@apikey_endpoint.route("/apikey/search", methods=["GET"])
+def list_apikey():
+    
+    session = db.Session()
+    user = session.query(User).filter(User.username.like(request.json['username'])).first()
+    if user is None:
+        return not_found()
+    
+    apikeys = session.query(APIKey).filter(APIKey.user_id == user.id).all()
+    
+    return success({'apikey_list': apikeys})
+    
+    
+@apikey_endpoint.route("/apikey", methods=["GET"])
+@exclude_from_val
+def lookup_apikey():
+    
+    if not validate_request(request, fields=['uuid']):
+        return bad_request()
+    
+    session = db.Session()
+    apikey = session.query(APIKey).filter(APIKey.id == request.json['apikey_uuid']).first()
+    
+    if apikey is None:
+        return not_found()
+    else:
+        valid_till = datetime.strftime(apikey.expiry_date, "%Y-%m-%d")
+        return success({'uuid': apikey.id, 'name': apikey.name, 'valid_till': valid_till, 'valid': not apikey.disabled})
+    
+    
+
+@apikey_endpoint.route("/apikey/revoke", methods=["PUT"])
+def revoke_service():
+    session = db.Session()
+    user = session.query(User).filter(User.username == request.json['username']).first()
+    apikey = session.query(APIKey).filter(APIKey.user_id == user.id and APIKey.id == request.json['apikey_uuid']).first()
+    if apikey is None:
+        return not_found()
+    else:
+        apikey.disabled = True
+        session.commit()
+        return success({'uuid': apikey.id, 'is_valid': not apikey.disabled})
+    
+
+@apikey_endpoint.before_request
+def validate():
+    if not validate_request(request):
+        return bad_request()
+
+
+
+    
```

Comment:
Leave one empty line at the end of the file.

Reply:
Please do not mark comments as resolved - especially when the suggested change is not addressed. Instead of leaving five lines at the end of the line - please leave just one line.